internal/npm_install/npm_install.bzl

```diff
@@ -427,6 +434,20 @@ yarn_install = repository_rule(
 See yarn CLI docs https://yarnpkg.com/en/docs/cli/install for complete list of supported arguments.""",
             default = [],
         ),
+        "frozen_lockfile": attr.bool(
+            default = True,
+            doc = """Use the `--frozen-lockfile` flag for yarn.
+
+Don’t generate a `yarn.lock` lockfile and fail if an update is needed.
+
+This flag enables an exact install of the version that is specified in the `yarn.lock`
+file. This helps to have reproduceable builds across builds.
+
+To update a dependency or install a new one run the `yarn install` command with the
+vendored yarn binary. `bazel run @nodejs//:yarn install`. You can pass the options like
```

Comment:
You've written here: use vendored yarn binary to update dependency. But it is impossible to do because running vendored yarn binary now requires actual yarn.lock firstly lmao